packages/openapi-to-graphql/src/schema_builder.ts

```diff
@@ -1026,80 +1026,63 @@ function linkOpRefToOpId<TSource, TContext, TArgs>({
 }
 
 /**
- * Determin if an argument should be created if the argument has already been
+ * Determine if an argument should be created if the argument has already been
  * provided through the options
  */
-function skipArg<TSource, TContext, TArgs>(
+function shouldSkipArgumentBecauseOptions<TSource, TContext, TArgs>(
   parameter: ParameterObject,
   operation: Operation,
   data: PreprocessingData<TSource, TContext, TArgs>
 ): boolean {
-  if (typeof data.options === 'object') {
-    switch (parameter.in) {
-      case 'header':
-        // Check header option
-        if (
-          typeof data.options.headers === 'object' &&
-          parameter.name in data.options.headers
-        ) {
-          return true
-        } else if (typeof data.options.headers === 'function') {
-          const headers = data.options.headers(
-            operation.method,
-            operation.path,
-            operation.oas.info.title
-          )
-
-          if (typeof headers === 'object') {
-            return true
-          }
+  // Get the value of `requestOptions` as it can be both an object or a function
+  const requestOptionsValue =
+    typeof data.options.requestOptions === 'object'
+      ? data.options.requestOptions
+      : typeof data.options.requestOptions === 'function'
+      ? data.options.requestOptions(
```

Comment:
Here we only provide `method`, `path`, and `title`. No resolver arguments or generated request options.